index.js

```diff
@@ -1,14 +1,33 @@
 'use strict'
 
 const fp = require('fastify-plugin')
-const { performance } = require('perf_hooks')
+const { performance, PerformanceObserver } = require('perf_hooks')
 const summary = require('summary')
 
 const ONSEND = 'on-send-'
 const PREHANDLER = 'pre-handler-'
 const ROUTES = 'fastify-routes:'
+let observedEntries = []
 
 module.exports = fp(async function (fastify, opts) {
+  const obs = new PerformanceObserver((items) => {
+    const fetchedItems = items.getEntries()
+    for (let i = 0; i < fetchedItems.length; i++) {
+      const e = fetchedItems[i]
+      if (e.name.indexOf(ROUTES) === 0) {
+        const key = e.name.split(':')[1]
+        if (observedEntries[key]) {
```

Comment:
it seems we are using `observedEntries` like an object, but it is an array instead.

Reply:
Yes, thanks, it started as an Array then moved to object, I'll fix that.